<launch>

  <arg name="scale" default="700" />
  <arg name="port" default="4003" />
  <arg name="stdout" default="screen" />
  <arg name="net_name" default="didi2p" />

  <param name="/birdview/grid_dim" value="35" />
  <param name="/birdview/cell_size" value="0.05" />

  <include file="$(find didi_challenge)/launch/laser_birdview.launch">
    <!-- <param name="grid_dim" value="35" />
    <param name="cell_size" value="0.05" /> -->
    <param name="cell_size_height_map" default="0.25" />
  </include>

  <!-- OBJECT DETECTION BRANCH -->
  <group ns="cnn">
    <param name="port" value="$(arg port)" />
    <!-- First stage -->
    <node pkg="cnn_detection" type="image_sender_idx" name="image_sender" output="$(arg stdout)" required="true">
      <remap from="/stereo_camera/left/image_rect_color" to="/bird_view"/>
    </node>

    <!-- Faster R-CNN classifier - Modified for ROIs inclusion-->
    <node pkg="cnn_detection" type="stream.py" name="stream_classify" output="$(arg stdout)" required="true">
      <param name="~use_external_rois" value="False" />
      <param name="~scale" value="$(arg scale)" />
      <param name="~net" value="$(arg net_name)" />
    </node>

    <!-- NMS-->
    <node pkg="cnn_detection" type="nms_detections.py" name="nms" output="$(arg stdout)" required="true">
      <param name="~y_offset" value="0" />
      <param name="~net" value="$(arg net_name)" />
    </node>
  </group>

  <include file="$(find didi_challenge)/launch/smart_tracker.launch">
    <param name="max_missings" value="100" />
    <param name="min_detections" value="1" />
  </include>
  <include file="$(find didi_challenge)/launch/interpolation.launch" />
  <include file="$(find didi_challenge)/launch/tracklet_gen.launch" />
</launch>
